File: api/src/main/kotlin/pl/bas/microtwitter/dto/ProfileUpdateDTO.kt
package pl.bas.microtwitter.dto

data class ProfileUpdateDTO(
        var fullName: String?,
        var description: String?,
        var location: String?,
        var profileLinkColor: String?,
        var url: String?,
        var birthdate: BirthdateDTO?,
        var avatarUrl: String?,
        var backgroundUrl: String?
)

data class BirthdateDTO(
        var day: Short?,
        var month: Short?,
        var year: Short?
)

File: ui/src/app/dto/ProfileUpdateDTO.ts
export interface ProfileUpdateDTO {
  fullName?: string;
  description?: string;
  location?: string;
  profileLinkColor?: string;
  url?: string;
  birthdate?: BirthdateDTO;
  avatarUrl ?: string;
  backgroundUrl ?: string;
}

export interface BirthdateDTO {
  day?: number;
  month?: number;
  year?: number;
}
